### PR TITLE
roachtest: add a new cdc/kafka-chaos-single-row test

### PR DESCRIPTION
**Pull request by andyyang890**
This patch adds a new `cdc/kafka-chaos-single-row` roachtest, which
will serve as a regression test for the Kafka restarts bug. It is
currently configured to only be able to run manually since we have not
yet fixed the bug.

Informs #122372

Release note: None

### CODE REVIEW

**Comment by andyyang890** [comment]
This PR depends on #124195

**Review by rharding6373 — APPROVED** [review]
:lgtm: Nice test!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/124313#-:-Ny3UvE313Brb2Cn-xeR:bcl2h3v)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @andyyang890, @asg0451, @nameisbhaskar, and @vidit-bhat)

___
*[`pkg/cmd/roachtest/tests/cdc.go` line 1578 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/124313#-Ny3TD566q55D7s6h66b:-Ny3TD566q55D7s6h66c:bi0ea4o) ([raw file](https://github.com/cockroachdb/cockroach/blob/0bd0c440a13c44e12a92c37725fda5b7aace60fb/pkg/cmd/roachtest/tests/cdc.go#L1578)):*
> ```Go
> 		Leases:           registry.MetamorphicLeases,
> 		CompatibleClouds: registry.AllExceptAWS,
> 		// TODO(yang): Add this to the nightly test suite after we fix the Kafka restart bug.
> ```

nit: s/yang/#122372
___
*[`pkg/cmd/roachtest/tests/cdc.go` line 1626 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/124313#-Ny3TraD7iuUx77wXzqA:-Ny3TraD7iuUx77wXzqB:b-s2biut) ([raw file](https://github.com/cockroachdb/cockroach/blob/0bd0c440a13c44e12a92c37725fda5b7aace60fb/pkg/cmd/roachtest/tests/cdc.go#L1626)):*
> ```Go
> 				stmt := fmt.Sprintf(`UPDATE t SET x = %d WHERE id = 1;`, i)
> 				if i == 0 {
> 					stmt = fmt.Sprintf(`INSERT INTO t VALUES (1, %d);`, i)
> ```

Consider lifting the insert statement and its execution out of the loop.

**Comment by rharding6373** [comment]
Please wait for @asg0451 to have a chance to review before merging. Thanks!

**Review by andyyang890 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/124313)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @asg0451, @nameisbhaskar, @rharding6373, and @vidit-bhat)

___
*[`pkg/cmd/roachtest/tests/cdc.go` line 1578 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/124313#-Ny3TD566q55D7s6h66b:-Ny3otGIDaGBg97X-N2f:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/0bd0c440a13c44e12a92c37725fda5b7aace60fb/pkg/cmd/roachtest/tests/cdc.go#L1578)):*
<details><summary><i>Previously, rharding6373 (Rachael Harding) wrote…</i></summary><blockquote>

nit: s/yang/#122372
</blockquote></details>

Done.
___
*[`pkg/cmd/roachtest/tests/cdc.go` line 1626 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/124313#-Ny3TraD7iuUx77wXzqA:-Ny3orEzA6FR4771zNDt:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/0bd0c440a13c44e12a92c37725fda5b7aace60fb/pkg/cmd/roachtest/tests/cdc.go#L1626)):*
<details><summary><i>Previously, rharding6373 (Rachael Harding) wrote…</i></summary><blockquote>

Consider lifting the insert statement and its execution out of the loop.
</blockquote></details>

Done.

**Review by asg0451 — APPROVED** [review]
> Reviewed 3 of 4 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/124313)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @nameisbhaskar, @rharding6373, and @vidit-bhat)

**Comment by andyyang890** [comment]
TFTRs!

bors r=rharding6373,asg0451

**Pull request merged**
